torch/csrc/dynamo/eval_frame.c

```diff
@@ -819,6 +819,24 @@ static PyObject* set_eval_frame_py(PyObject* dummy, PyObject* callback) {
   return set_eval_frame(callback, PyThreadState_GET());
 }
 
+static PyObject* set_eval_frame_callback_enabled(PyObject* dummy, PyObject* enabled) {
+  bool prior = eval_frame_callback_enabled_get();
+  if (enabled == Py_True) {
+    eval_frame_callback_enabled_set(true);
+  } else if (enabled == Py_False) {
+    eval_frame_callback_enabled_set(false);
+  } else {
+    DEBUG_TRACE0("arg error");
+    PyErr_SetString(PyExc_TypeError, "expected bool");
+    return NULL;
+  }
+  return PyBool_FromLong(prior);
+}
```

Comment:
Doesn't this do the same thing as `set_eval_frame(None)` (which disables dynamo and makes everything run eagerly).

We also have `TORCHDYNAMO_DISABLE=1` which does something similar.

The one advantage this has is it is thread local, but IMO we should just fix the thread safety of the existing stuff rather than adding a duplicate API.

Reply:
`eval_frame_callback_enabled_get` is higher priority than `set_eval_frame`. The new API is useful for cases where we want to ignore future `torch.compile` invocations, which would overwrite the eval frame handler.

We have a lot of discussion over at https://github.com/pytorch/pytorch/pull/132926.

Reply:
It would be much faster to disable frame evaluation entirely (do the check once when we hit torch.compiled code), since then the CPython JIT can kick in.  We also won't need to do an extra check on every function call.